pkg/resources/scim_integration.go

```diff
@@ -74,140 +86,170 @@ func SCIMIntegration() *schema.Resource {
-	// We need to grab them in a loop
-	var k, pType string
-	var v, unused interface{}
-	stmt = snowflake.NewSCIMIntegrationBuilder(id).Describe()
-	rows, err := db.Query(stmt)
-	if err != nil {
-		return fmt.Errorf("could not describe security integration")
+	if err := d.Set("enabled", integration.Enabled); err != nil {
+		return diag.FromErr(err)
 	}
-	defer rows.Close()
-	for rows.Next() {
-		if err := rows.Scan(&k, &pType, &v, &unused); err != nil {
-			return fmt.Errorf("unable to parse security integration rows")
+	typeParts := strings.Split(integration.IntegrationType, "-")
+	if len(typeParts) < 2 {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Invalid security integration type format.",
+				Detail:   fmt.Sprintf("expected \"SCIM - <scim_client>\", got: %s", integration.IntegrationType),
+			},
 		}
-		switch k {
+	}
+	scimClient := strings.TrimSpace(typeParts[1])
+	if err := d.Set("scim_client", scimClient); err != nil {
+		return diag.FromErr(err)
+	}
+	integrationProperties, err := client.SecurityIntegrations.Describe(ctx, id)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	for _, property := range integrationProperties {
+		name := property.Name
+		value := property.Value
+		switch name {
+		case "ENABLED", "COMMENT":
+			// We set this using the SHOW INTEGRATION call so let's ignore it here
 		case "NETWORK_POLICY":
-			if err := d.Set("network_policy", v.(string)); err != nil {
-				return fmt.Errorf("unable to set network policy for security integration")
+			if err := d.Set("network_policy", value); err != nil {
+				return diag.FromErr(err)
+			}
+		case "SYNC_PASSWORD":
+			if err := d.Set("sync_password", helpers.StringToBool(value)); err != nil {
+				return diag.FromErr(err)
 			}
 		case "RUN_AS_ROLE":
-			if err := d.Set("provisioner_role", v.(string)); err != nil {
-				return fmt.Errorf("unable to set provisioner role for security integration")
+			if err := d.Set("run_as_role", value); err != nil {
+				return diag.FromErr(err)
 			}
 		default:
-			log.Printf("[WARN] unexpected security integration property %v returned from Snowflake", k)
+			log.Printf("[WARN] unexpected property %v returned from Snowflake", name)
 		}
 	}
 
-	return err
+	return nil
 }
 
-// UpdateSCIMIntegration implements schema.UpdateFunc.
-func UpdateSCIMIntegration(d *schema.ResourceData, meta interface{}) error {
+// UpdateContextSCIMIntegration implements schema.UpdateFunc.
+func UpdateContextSCIMIntegration(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*provider.Context).Client
-	db := client.GetConn().DB
-	id := d.Id()
-
-	stmt := snowflake.NewSCIMIntegrationBuilder(id).Alter()
-
-	var runSetStatement bool
-
-	if d.HasChange("scim_client") {
-		runSetStatement = true
-		stmt.SetString(`SCIM_CLIENT`, d.Get("scim_client").(string))
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
+	var runSet, runUnset bool
```

Comment:
You can do as in e.g. secondary_database, that instead of setting flags you can compare `set` with `sdk.NewScimIntegrationSetRequest()` to see if any of the fields has been set.

Reply:
Done.